### PR TITLE
(BKR-71) MSI from URL

### PR DESCRIPTION
**Pull request by DLuCJ**
* Previously, install_puppet_agent_dev_repo_on and
   install_puppet_agent_pe_promoted_repo_on installed MSIs on
   Windows agents by downloading an MSI to the coordinator and
   copying it to each agent.

   It is now possible to pass a URL directly to the install_msi_on.
   Doing so avoids unnecessary copying of the MSI.

* This change builds up a URL from strings returned by:
   host.puppet_agent_dev_package_info
   host.pe_puppet_agent_promoted_package_info

   It then passes the URL to install_msi_on.

* Additionally, it removes the curl / Powershell
   invocations in install_a_puppet_msi_on, as install_msi_on
   replaces that functionality.

*  This is related to changes in https://github.com/puppetlabs/beaker-pe/pull/67

### CODE REVIEW

**Comment by DLuCJ** [comment]
@Iristyle @kevpl  
This likely needs some work, but as of three weeks ago, passed PE/regular acceptance testing for 
CentOS/RH master -> CentOS/RH agent, and CentOS/RH master -> 2012R2 agent using my machine as coordinator. 

PE acceptance passed for these paths by modifying PE ver:
https://github.com/puppetlabs/beaker-pe/blob/master/lib/beaker-pe/install/pe_utils.rb#L532
https://github.com/puppetlabs/beaker-pe/blob/master/lib/beaker-pe/install/pe_utils.rb#L545

Spec modifications may need to be rethought.

File: lib/beaker/dsl/install_utils/foss_utils.rb [hunk 1]
```diff
@@ -1089,7 +1085,7 @@ def install_packages_from_local_dev_repo( host, package_name )
         #
         # @return nil
         def install_puppet_agent_dev_repo_on( hosts, opts )
-
+          # DLU: require 'pry'; binding.pry
```
Comment (kevpl) [inline lib/beaker/dsl/install_utils/foss_utils.rb]: @DLuCJ these pry comments should be deleted

File: lib/beaker/dsl/install_utils/foss_utils.rb [hunk 2]
```diff
@@ -584,6 +585,7 @@ def install_puppet_agent_from_msi_on(hosts, opts)
             end
 
             install_a_puppet_msi_on(host, opts)
+            # DLU: install_puppet_agent_dev_repo_on(hosts, opts)
```
Comment (kevpl) [inline lib/beaker/dsl/install_utils/foss_utils.rb]: @DLuCJ what purpose do these comments serve?

**Comment by kevpl** [comment]
@DLuCJ last comment is still current

**Comment by kevpl** [comment]
@DLuCJ last comment is still current. If this ticket isn't moved forward in a week, it'll be closed due to inactivity.

**Pull request closed without merging**